tedana/utils/masking.py

```diff
@@ -0,0 +1,125 @@
+"""
+"""
+import numpy as np
+import nibabel as nib
+
+
+def apply_mask_nii_me(imgs, mask_img):
+    """
+    Apply multi-echo mask to set of nifti images. Mask may vary by echo.
+    """
+    pass
+
+
+def apply_mask_gii_me(img, mask_img):
+    """
+    Apply multi-echo mask to set of gifti images. Mask may vary by echo.
+    TODO: Extend to work with multiple images.
+
+    Parameters
+    ----------
+    img : :obj:`nibabel.gifti.gifti.GiftiImage`
+        3D (Vertex, Echo, Time) gifti image containing data.
+    mask_img : :obj:`nibabel.gifti.gifti.GiftiImage`
+        1D (Vertex) or 2D (Vertex, Echo) gifti image with boolean or bool-like
+        values.
+
+    Returns
+    -------
+    masked_arr : :obj:`numpy.ndarray`
+        M x E x T data.
+    """
+    if not isinstance(mask_img, nib.gifti.GiftiImage):
+        raise TypeError('Provided mask is not a GiftiImage.')
+    elif len(mask_img.darrays) > 1:
+        raise ValueError('Mask may only contain one GiftiDataArray.')
+
+    if not isinstance(img, nib.gifti.GiftiImage):
+        raise TypeError('Provided data img is not a GiftiImage.')
+
+    img_arrs = [np.array(darr.data) for darr in img.darrays]
+    all_imgs_same = np.all(arr.shape == img_arrs[0].shape for arr in img_arrs)
```

Comment:
You'll need a list comprehension here otherwise you get a generator: `all_imgs_same = np.all([arr.shape == img_arrs[0].shape for arr in img_arrs])`.

Reply:
Thx. Nice catch!